waveform_editor/configuration.py

```diff
@@ -139,8 +188,36 @@ def remove_group(self, path):
             path: A list representing the path to the group to be removed.
         """
         parent_group = self if len(path) == 1 else self.traverse(path[:-1])
-        group = parent_group.groups.pop(path[-1])
+        group = parent_group.groups[path[-1]]
+
+        to_remove = self._collect_waveforms_in_group(group)
+
+        for wf_name, grp in self.waveform_map.items():
+            if wf_name not in to_remove:
+                for wf in grp.waveforms.values():
+                    if isinstance(wf, DerivedWaveform) and to_remove.intersection(
+                        wf.dependent_waveforms
+                    ):
+                        raise RuntimeError(
+                            f"Cannot remove group {group.name}. "
+                            f"{wf.name!r} depends on waveform in it."
+                        )
+
+        del parent_group.groups[path[-1]]
         self._recursive_remove_waveforms(group)
+        self._calculate_bounds()
+        for wf in list(self.waveform_map.keys()):
+            if wf not in self.waveform_map:
+                self.dependency_graph.remove_node(wf)
+
+    def _collect_waveforms_in_group(self, group):
+        waveforms = set()
```

Comment:
Let's use a list here
```suggestion
        waveforms = []
```